reVX/handlers/xmission_cost_layers.py

```diff
@@ -13,3 +15,89 @@ class XmissionCostsLayers(ExclusionLayers):
     """
     Handler for Transmission cost layers
     """
+    def __init__(self, h5_file, capacity, hsds=False):
+        """
+        Parameters
+        ----------
+        h5_file : str | list | tuple
+            .h5 file containing exclusion layers and techmap,
+            or a list of h5 files
+        capacity : int
+            Capacity bin to extract costs for
+        hsds : bool, optional
+            Boolean flag to use h5pyd to handle .h5 'files' hosted on AWS
+            behind HSDS, by default None
+        """
+        self._capacity = capacity
+        super().__init__(h5_file, hsds=hsds)
+
+    def __getitem__(self, keys):
+        ds, ds_slice = parse_keys(keys)
+
+        if ds.lower().startswith('lat'):
+            out = self._get_latitude(*ds_slice)
+        elif ds.lower().startswith('lon'):
+            out = self._get_longitude(*ds_slice)
+        elif ds.lower() == 'costs':
+            out = self._get_costs(*ds_slice)
+        elif ds.lower() == 'mcp_costs':
+            out = self._get_mcp_costs(*ds_slice)
+        else:
+            out = self._get_layer(ds, *ds_slice)
+
+        return out
+
+    def _get_costs(self, *ds_slice):
+        """
+        Compute the raw tie-line costs for the given slice
+
+        Parameters
+        ----------
+        ds_slice : tuple of int | list | slice
+            tuple describing slice of cost array to extract
+
+        Returns
+        -------
+        costs : ndarray
+            Array of raw tie-line costs
+        """
+        layers = ['base_costs', 'multipliers_{}mw'.format(self._capacity)]
+        for layer_name in layers:
+            if layer_name not in self.layers:
```

Comment:
Why would a layer not exist? Not created yet?

Reply:
Exactly, not created yet. My long term goal for all of this code is to make it available to the public and its a very quick simple check to just make sure the layer exists. we also raise a more verbose error than the basic key error you'd get form h5py by listing all available layers. I'll add more comments that the layers are needed to compute...